File: scrooge-generator/src/main/resources/javagen/readList.java
TList _list_{{name}} = _iprot.readListBegin();
{{name}} = new ArrayList<{{eltType}}>();
int _i_{{name}} = 0;
{{eltType}} {{eltName}};
while (_i_{{name}} < _list_{{name}}.size) {
{{#eltReadWriteInfo}}
  {{>readValue}}
{{/eltReadWriteInfo}}
  {{name}}.add({{eltName}});
  _i_{{name}} += 1;
}
_iprot.readListEnd();

File: scrooge-generator/src/main/resources/cocoagen/readList.m
int _{{name}}_size;
[inProtocol readListBeginReturningElementType:NULL size:&_{{name}}_size];
NSMutableArray *{{name}}_mutable = [[NSMutableArray alloc] initWithCapacity:_{{name}}_size];
for (int _{{name}}_i = 0; _{{name}}_i < _{{name}}_size; ++_{{name}}_i) {
    {{eltType}}{{#eltReadWriteInfo}}{{#isStruct}} *{{/isStruct}}{{/eltReadWriteInfo}} {{eltName}};
{{#eltReadWriteInfo}}
    {{>readValue}}
{{/eltReadWriteInfo}}
    [{{name}}_mutable addObject: {{eltName}}];
}
{{name}} = {{name}}_mutable;
[inProtocol readListEnd];